packages/gatsby-source-wordpress/src/steps/process-and-validate-plugin-options.ts

```diff
@@ -15,7 +15,32 @@ interface IOptionsProcessor {
   processor: (options: IProcessorOptions) => IPluginOptions | void
 }
 
+let warnedAboutMediaItemLazyNodes = false
+
 const optionsProcessors: Array<IOptionsProcessor> = [
+  {
+    name: `MediaItem.lazyNodes doesn't work in Gatsby v4+`,
+    test: ({ userPluginOptions }): boolean =>
+      userPluginOptions?.type?.MediaItem?.lazyNodes,
+    processor: ({ helpers, userPluginOptions }): IPluginOptions => {
+      if (usingGatsbyV4OrGreater) {
+        helpers.reporter.panic(
+          formatLogMessage(
+            `The type.MediaItem.lazyNodes option isn't supported in Gatsby v4+ due to query running using JS workers in PQR (Parallell Query Running). lazyNodes creates nodes in GraphQL resolvers and PQR doesn't support that.\n\nIf you would like to prevent gatsby-source-wordpress from fetching File nodes for each MediaItem node, set the type.MediaItem.createFileNodes option to false.`
+          )
+        )
+      } else if (!warnedAboutMediaItemLazyNodes) {
+        warnedAboutMediaItemLazyNodes = true
+        helpers.reporter.warn(
+          formatLogMessage(
+            `\nThe type.MediaItem.lazyNodes option wont be supported in Gatsby v4+ due to query running using JS workers in PQR (Parallell Query Running). lazyNodes creates nodes in GraphQL resolvers and PQR doesn't support that.\n\nThis option works with your current version of Gatsby but will stop working in Gatsby v4+.\n\nIf you would like to prevent gatsby-source-wordpress from fetching File nodes for each MediaItem node, set the type.MediaItem.createFileNodes option to false instead.\n`
+          )
+        )
+      }
+
+      return userPluginOptions
+    },
+  },
```

Comment:
Can we force it to be false when `_CFLAGS_.GATSBY_MAJOR === '4'`? this would allow us to drop the check everywhere else.

Reply:
I don't want to force it to be false because it's being used elsewhere in the code to do things that don't seem to depend on the gatsby version. Might be better to leave it as is and bring this up to Tyler when he gets back.

Reply:
That's fine, was more a suggestion to keep things nice and tidy :)